airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/strip/pipeline.py

```diff
@@ -0,0 +1,195 @@
+          stdout=f"The connector has unrecognized source files: {self.invalid_files}",
+        )
+      
+      # Detect connector class name to make sure it's inherited from source declarative manifest 
+      # and does not override `streams` method
+      connector_source_py = (connector_source_code_dir / "source.py").read_text()
+
+      if "YamlDeclarativeSource" not in connector_source_py:
+        return StepResult(
+          step=self,
+          status=StepStatus.SKIPPED,
+          stdout="The connector does not use the YamlDeclarativeSource class.",
+        )
+      
+      if "def streams" in connector_source_py:
+        return StepResult(
+          step=self,
+          status=StepStatus.SKIPPED,
+          stdout="The connector overrides the streams method.",
+        )
+
+      return StepResult(
+        step=self, 
+        status=StepStatus.SUCCESS, 
+        stdout=f"{self.context.connector.technical_name} is a valid candidate for migration.")
+
+class StripConnector(Step):
+    context: ConnectorContext
+
+    title = "Strip the connector to manifest-only."
+
+    async def _run(self) -> StepResult:
+
+      # 1. Move manifest.yaml to the root level of the directory
+      self.logger.info(f"Moving manifest to the root level of the directory")
+      connector_source_code_dir = self.context.connector.code_directory / self.context.connector.technical_name.replace("-", "_")
+
+      manifest_file = connector_source_code_dir / "manifest.yaml"
+      manifest_file = manifest_file.rename(self.context.connector.code_directory / "manifest.yaml")
+
+      if manifest_file not in self.context.connector.code_directory.iterdir():
+        return StepResult(
+          step=self,
+          status=StepStatus.FAILURE,
+          stdout="Failed to move manifest.yaml to the root level of the directory."
+        )
+      
+      # We don't want to delete the source_<name> folder
+      FILES_TO_LEAVE.append(self.context.connector.technical_name.replace("-", "_"))
+
+      # 2. Delete everything that is not in an allow-list of files
+      for file in self.context.connector.code_directory.iterdir():
+        if file.name not in FILES_TO_LEAVE and not file.is_dir():
+          self.logger.info(f"Deleting {file.name}")
+          file.unlink()
+        elif file.name not in FILES_TO_LEAVE and file.is_dir():
+          self.logger.info(f"Deleting {file.name} folder")
+          shutil.rmtree(file)
+
+        if file in self.context.connector.code_directory.iterdir() and file.name not in FILES_TO_LEAVE:
+          return StepResult(
+            step=self,
+            status=StepStatus.FAILURE,
+            stdout=f"Failed to delete {file.name}"
+          )
+
+      # 3. Grab the cdk tag from metadata.yaml and update it
+      metadata_file = self.context.connector.code_directory / "metadata.yaml"
+      with open(metadata_file, "r") as file:
+        metadata = yaml.safe_load(file)
+        tags = metadata['data']['tags']
```

Comment:
We should add a `langage:manifest-only` tag, not a cdk tag. I don't know if we should delete the `cdk:low-code` and `language:low-code` though. /cc @bnchrch

Reply:
Haha I knew that too, my Friday afternoon brain musta kicked in 🫠 Switched to appending the language tag for now and can add the logic to delete the others if needed